src/components/implementation/no_interface/llbooter/comp_config.h

```diff
@@ -0,0 +1,66 @@
+#ifndef COMP_CONFIG_H
+#define COMP_CONFIG_H
+
+#include <cos_types.h>
+#include <consts.h>
+#include <string.h>
+
+struct cos_config_info_t *comp_configs[MAX_NUM_COMPS];
+
+enum
+{
+	SL_RUMPCOS = 0,
+	SHMEM,
+	SL_TEST_BOOT,
+};
+
+/*
+ * Define your component configuration here.
+ * The name of the cos_config_info struct should be that of the final object file
+ * for the component so other users can clearly associate components to their
+ * configurations.
+ *
+ * The first key-value pair should be the object name (same as the name you juse used),
+ * for the struct itself. The llbooter leverages this first key-value pair to match up
+ * object files to their appropriate components during runtime.
+ *
+ * After defining the remaining key-value pairs as you like, add your configuration struct
+ * to the master list which the llbooter will parse through.
+ * Add your own enum entry used to offset the config struct into the master list.
+ * This enum is only used within this header file.
+ *
+ * Use the below configuration for the Rump Kernel as an example.
+ */
+
+static struct cos_config_info_t sl_rumpcos;
+static struct cos_config_info_t shmem;
+static struct cos_config_info_t sl_test_boot;
+
+static void
+__init_configs()
```

Comment:
This is all kind of horrible: having to write code to add the configs.  Will we just have 10's of these files lying around, one per runscript?  I wonder if a patch that includes the code to get the KV store and all of that, *without* the ability to set the KVs is more valuable for now.  We really just want the loader to add to the KV.  A better option, perhaps:  use the current code for setting the initial arguments to set a "cmdline" key to the value passed in the runscript.  

This code is not where we want to end up, obviously.  I'm interested in figuring out what a decent transition plan to that is.